app/models/forms/mappings/Constraints.scala

```diff
@@ -18,8 +18,12 @@ package models.forms.mappings
 
 import play.api.data.validation.{Constraint, Invalid, Valid}
 
+import scala.util.matching.Regex
+
 trait Constraints {
 
+  private val emailRegex = """^[a-zA-Z0-9\.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$""".r
```

Comment:
Is there maybe an internal library for doing this already?

Reply:
there is a internal library but is not very robust and in case if business need specific errors in specific cases you can't do this in library, this is why we kept this regex in place here